### DISCUSSION TITLE
Adding support for Mistral-Nemo-Instruct

### DISCUSSION THREAD

**Discussion by handshape** [ggml-org/llama.cpp#8576] [Ideas]
Per the ticket template for feature requests, how does the community feel about adding support for Mistral-Nemo-Instruct?
On its face, it looks like there's a need to add support for a pre-tokenizer type called `mistral-bpe`.

**Comment by iamlemec** [comment]
I just added in the pre-tokenizer quickly and converted. But on running, you get errors related to incorrect tensor shapes. Namely, the embedding is `n_embd = 5120`, but this does not split evenly into the Q and K/V sizes like usual. So the `attn_q.weight` tensor is actually `4096 x 5120` instead of `5120 x 5120` and so on.

From the look of it, only Gemma and Gemma2 currently do something similar.

**Comment by handshape** [comment]
Looks like work is underway.
https://github.com/ggerganov/llama.cpp/issues/8577
